workers/unity/Packages/io.improbable.worker.sdk/EventTracing.cs

```diff
@@ -0,0 +1,403 @@
+
+        public bool IsEnabled { get; private set; }
+
+        public EventTracer(EventTracerParameters[] parameters)
+        {
+            unsafe
+            {
+                ParameterConversion.ConvertEventTracer(parameters, (internalParameters, handles) =>
+                {
+                    eventTracer = CEventTrace.EventTracerCreate(internalParameters);
+                    handleList = handles;
+                });
+            }
+        }
+
+
+        public void Dispose()
+        {
+            eventTracer.Dispose();
+            handleList.ForEach(handle => handle.Dispose());
+        }
+
+        public void Enable()
+        {
+            CEventTrace.EventTracerEnable(eventTracer);
+            IsEnabled = true;
+        }
+
+        public void Disable()
+        {
+            CEventTrace.EventTracerDisable(eventTracer);
+            IsEnabled = false;
+        }
+
+        public void SetActiveSpanId(SpanId spanId)
+        {
+            CEventTrace.EventTracerSetActiveSpanId(eventTracer, ParameterConversion.ConvertSpanId(spanId));
+        }
+
+        public void ClearActiveSpanId(SpanId spanId)
+        {
+            // Uncomment once Worker SDK updated to 14.8.0 (version where EventTracerClearActiveSpanId is added)
+            // CEventTrace.EventTracerClearActiveSpanId(eventTracer, ParameterConversion.ConvertSpanId(spanId));
+        }
+
+        public SpanId GetActiveSpanId()
+        {
+            var nativeSpanId = CEventTrace.EventTracerGetActiveSpanId(eventTracer);
+            var spanId = new SpanId();
+            unsafe
+            {
+                ApiInterop.Memcpy(spanId.Data, nativeSpanId.Data, SpanId.SpanIdSize);
+            }
+
+            return spanId;
+        }
+
+        // Returns the SpanId of the newly-created Span in the EventTracer
+        public SpanId AddSpan(SpanId[] causes)
+        {
+            unsafe
+            {
+                if (causes == null)
+                {
+                    return AddSpan();
+                }
+
+                var causeIds = new CEventTrace.SpanId[causes.Length];
+                for (var i = 0; i < causes.Length; i++)
+                {
+                    causeIds[i] = ParameterConversion.ConvertSpanId(causes[i]);
+                }
+
+                CEventTrace.SpanId createdSpanId;
+                fixed (CEventTrace.SpanId* fixedCauseIds = causeIds)
+                {
+                    createdSpanId = CEventTrace.EventTracerAddSpan(eventTracer, fixedCauseIds, (uint) causeIds.Length);
+                }
+
+                var newSpanId = new SpanId();
+                ApiInterop.Memcpy(newSpanId.Data, createdSpanId.Data, SpanId.SpanIdSize);
+
+                return newSpanId;
+            }
+        }
+
+        public SpanId AddSpan()
+        {
+            var newSpanId = new SpanId();
+            unsafe
+            {
+                var createdSpanId = CEventTrace.EventTracerAddSpan(eventTracer, null, 1);
```

Comment:
The cause count on this should be `0` rather than `1` no?

Reply:
Yes, that should be `0`